platform_umbrella/apps/common_core/lib/common_core/et/knative_report.ex

```diff
@@ -3,35 +3,28 @@ defmodule CommonCore.ET.KnativeReport do
 
   use CommonCore, :embedded_schema
 
+  import CommonCore.ET.ReportTools
+
   alias CommonCore.Ecto.Schema
   alias CommonCore.Resources.FieldAccessors
   alias CommonCore.StateSummary
-  alias CommonCore.StateSummary.FromKubeState
 
   batt_embedded_schema do
     field :pod_counts, :map
   end
 
   def new(%StateSummary{knative_services: services} = state_summary) do
+    pods_by_owner = count_pods_by(state_summary, &FieldAccessors.labeled_owner/1)
+
     pod_counts =
-      Map.new(services, fn service ->
-        {service.name, num_pods(service, state_summary)}
-      end)
+      Map.new(services, &count_by_service(&1, pods_by_owner))
 
-    Schema.schema_new(__MODULE__,
-      pod_counts: pod_counts
-    )
+    Schema.schema_new(__MODULE__, pod_counts: pod_counts)
   end
 
   def new(opts) do
     Schema.schema_new(__MODULE__, opts)
   end
 
-  defp num_pods(service, state_summary) do
-    pods = FromKubeState.all_resources(state_summary, :pod)
-
-    Enum.count(pods, fn pod ->
-      FieldAccessors.labeled_owner(pod) == service.id
-    end)
-  end
+  defp count_by_service(service, pods_by_owner), do: {service.name, Map.get(pods_by_owner, service.id, 0)}
```

Comment:
Just grab the count from the pre-computed map and use it.

Reply:
Same.